### PR TITLE
Document the operator maintenance

### LINKED ISSUE #12: Documentation
The project is now in the proof-of-concept stage, and lacks the documentation (beside the docstrings and examples).

It needs some normal hosted HTML docs (e.g. on ReadTheDocs).

* [x] Hosting configured.
* [x] Builds configured.

Specifically, some topics to not forget:

* [x] Typical patterns to solve typical problems (similar to examples). E.g., children object creation.
* [x] Deployment patterns: namespace isolation, deployment and RBAC-based service accounts 
 (minimum rules needed for the framework, not the general recommendations), etc. See #17.
* [x] Getting starting (or quick-start) guide (isn't it an example/01-minimal?).
* [x] The requirement for `**kwargs` or `**_` (for no-linting) for the forward-compatibility with the new keywords passed to the existing handlers by the new versions of the framework (part of the DSL).
* [x] The full list of all the available keywords now: `body`, `meta`, `spec`, `status`, `patch`, `logger`, `retry`, `diff`, `old`, `new`.
* [x] How the state is persistent in k8s itself, point out that the framework is stateless (same as the web servers).
* [x] What happens when the operator is killed during the handler, and how to avoid the duplicated side-effects (i.e. idempotent handlers).
* [x] What happens when the changes are applied when the operator is dead or fails permanently due to the bugs (the last-seen state preservation and comparison), i.e. that no changes are left behind unnoticed.
* [x] Strategies to isolate the operators to the namespaces; but still to have the cluster-wide operators (cluster CRDs or all-namespace monitoring, as e.g. Postgres operators, i.e. not already domain-specific ad-hoc solutions).
* [x] Alternatives: CoreOS Operator Framework, all Go-based solutions, etc). Differences and similarities, and the self-positioning of Kopf.

### LINKED ISSUE #75: Add warning to docs about finalizers.
FYI. You may want to add a warning to the documentation about problems that can occur due to the way in which finalizers are used on a custom CRD. In particular, if the operator is deployed to a namespace and setup in standalone mode to monitor just the same namespace, if the operator is deleted before the custom CRD objects, then when the namespace is deleted, it will get stuck in terminating state.

This is especially problematic, because deletion of a project doesn't really guarantee any order of deletion where owner references and blocking conditions haven't been specified. Thus deletion of a namespace can result in the operator being deleted first, without there being a need to have explicitly delete the operator first to cause this issue. Even if the operator is deployed in a separate project, same issue can arise were the operator were separately deleted.

The only solution to this is to identify what the custom CRD object is that is blocking deletion of the namespace, and manually edit it to remove the finalizer. I don't know of any better solution at the moment, or how you could rework kopf to not be affected by the issue.

### PR DESCRIPTION
**Pull request by nolar**
> Issue : closes #75, closes #12

Add some pages for the maintenance of the Kopf-based operators, describing their behaviour on downtime, on restarts, how they keep the persistence.

Add the troubleshooting page for some unexpected behaviour, to make it clear (e.g. #75).

_A new section was needed, as this kind of information does not fit to Tutorial, Features, About Kopf, and not even to the Developers Manual (which is about developing Kopf itself, not the Kopf-based operators)._

Preview:

* https://kopf-nolars-fork.readthedocs.io/en/docs-maintenance/continuity.html
* https://kopf-nolars-fork.readthedocs.io/en/docs-maintenance/idempotence.html
* https://kopf-nolars-fork.readthedocs.io/en/docs-maintenance/troubleshooting.html

### CODE REVIEW

File: docs/continuity.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Kopf does not have any database. It stores all the information directly
+on the objects in the Kubernetes cluster (which means `etcd` usually).
+All information is retrieved and stored via the Kubernetes API.
+
+Specifically:
+
+* The cross-operator exchange is performed via the peering objects.
+  See :doc:`peering` for more info.
+* The last handled state of the object is stored in ``metadata.annotations``.
+  It is used for the diff calculation on the changes.
```
Comment (samurang87) [inline docs/continuity.rst]: "it is used to calculate diffs upon changes" ... I guess

File: docs/continuity.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  in ``status.kopf.progress`` (with ``status.kopf`` reserved for any
+  framework-related information in the future).
+
+
+Restarts
+========
+
+It is safe to kill the operator's pod (or process), and allow it to restart.
+
+The handlers that succeeded previously, will not be re-executed.
```
Comment (samurang87) [inline docs/continuity.rst]: remove the comma :)

File: docs/continuity.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+The handlers that did not execute yet, or were scheduled for retrying,
+will be retried by a new operators pod/process from the point where
+the old pod/process was terminated.
+
+The restart of an operator will only affect the handlers currently being
```
Comment (samurang87) [inline docs/continuity.rst]: Restarting an operator...

File: docs/continuity.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+========
+
+If the operator is down and not running, any changes to the objects
+are ignored and not handled. They will be handled when the operator starts:
+every time a Kopf-based operator starts, it lists all objects of the served
+resource kind, and checks for their state; if the state has changed since
+the object was last handled (no matter how long time ago),
+a new handling cycle starts.
+
+Only the last state is taken into account. All the intermenient changes
```
Comment (samurang87) [inline docs/continuity.rst]: I guess you have a typo here (intermediate? intervening?)

File: docs/continuity.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+are accumulated and handled together.
+This corresponds to the Kubernetes'es concept of eventual consistency
```
Comment (samurang87) [inline docs/continuity.rst]: This corresponds to Kubernetes's (🤔 ) concept of ...

File: docs/idempotence.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
+===========
+Idempotence
+===========
+
+Kopf provides the tools to make the handlers idempotent.
```
Comment (samurang87) [inline docs/idempotence.rst]: "Kopf provides tools..."

**Comment by nolar** [comment]
@samurang87 I've fixed some grammar issues :-)

**Pull request merged**